sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -568,8 +574,12 @@ object DataSource extends Logging {
     "org.apache.spark.Logging")
 
   /** Given a provider name, look up the data source class definition. */
-  def lookupDataSource(provider: String): Class[_] = {
-    val provider1 = backwardCompatibilityMap.getOrElse(provider, provider)
+  def lookupDataSource(conf: SQLConf, provider: String): Class[_] = {
```

Comment:
After more thinking, I think it don't worth to pass the whole SQLConf into this function, we just need to know whether `SQLConf.ORC_USE_NEW_VERSION` is enabled.

Reply:
So, are you suggesting `lookupDataSource(provider, useNewOrc=true)`, @jiangxb1987 ?